### PR TITLE
Update swarmkit, add support for external ca

### PR DESCRIPTION
**Pull request by tonistiigi**
- Update vendoring for swarmkit, libnetwork, engine-api
- Add support for setting external CAs in `swarm init/update`

~~Do not merge before #23584 , that also vendors newer engine-api and will conflict. I'll fix the conflicts in this pr after #23584 is merged.~~

@mavenugo @tiborvass @aaronlehmann

### CODE REVIEW

**Comment by mavenugo** [comment]
Thanks for vendoring in libnetwork & swarmkit.

LGTM

File: docs/reference/api/docker_remote_api_v1.25.md [hunk 1]
```diff
@@ -3645,8 +3645,15 @@ JSON Parameters:
...
-    - **CAConfig** – CA configuration.
+    - **CAConfig** – Certificate authority configuration.
         - **NodeCertExpiry** – Automatic expiry for nodes certificates.
+        - **ExternalCA** - Configuration for forwarding signing requests to an external
+          certificate authority.
+            - **Protocol** - Protocol for communication with the external CA
+              (currently only "cfssl" is supported).
+            - **URL** - URL where certificate signing requests should be sent.
+            - **Options** - An object with key/value pairs that are interpreted
+              as protocol-specific options for the external CA driver.
```
Comment (thaJeztah) [inline docs/reference/api/docker_remote_api_v1.25.md]: These changes should also be in the 1.24 docs (docker 1.12 is API version 1.24)

Comment (aaronlehmann) [inline docs/reference/api/docker_remote_api_v1.25.md]: @thaJeztah: Fixed, thanks.

**Comment by stevvooe** [comment]
LGTM

**Comment by nishanttotla** [comment]
LGTM

**Comment by thaJeztah** [comment]
LGTM

**Pull request merged**